server/app/controllers/applicant/RedirectController.java

```diff
@@ -107,32 +111,35 @@ public CompletionStage<Result> programBySlug(Http.Request request, String progra
                         // for this program, redirect to program version associated
                         // with that application if so.
                         if (programForExistingApplication.isPresent()) {
+                          long programId = programForExistingApplication.get().id();
                           return CompletableFuture.completedFuture(
-                              redirect(
-                                      controllers.applicant.routes.ApplicantProgramReviewController
-                                          .review(
-                                              applicantId,
-                                              programForExistingApplication.get().id()))
-                                  .flashing("redirected-from-program-slug", programSlug));
+                              goToActiveProgram(programId, applicantId, programSlug, request));
+                        } else {
+                          return programService
+                              .getActiveProgramDefinitionAsync(programSlug)
+                              .thenApply(
+                                  activeProgramDefinition ->
+                                      goToActiveProgram(
+                                          activeProgramDefinition.id(),
+                                          applicantId,
+                                          programSlug,
+                                          request));
                         }
-
-                        return redirectToActiveProgram(applicantId, programSlug);
                       },
                       httpContext.current());
             },
             httpContext.current());
   }
 
-  private CompletionStage<Result> redirectToActiveProgram(long applicantId, String programSlug) {
-    return programService
-        .getActiveProgramDefinitionAsync(programSlug)
-        .thenApplyAsync(
-            (activeProgramDefinition) ->
-                redirect(
-                        controllers.applicant.routes.ApplicantProgramReviewController.review(
-                            applicantId, activeProgramDefinition.id()))
-                    .flashing("redirected-from-program-slug", programSlug),
-            httpContext.current());
+  private Result goToActiveProgram(
+      long programId, long applicantId, String programSlug, Http.Request request) {
+    return redirect(
+            controllers.applicant.routes.ApplicantProgramReviewController.review(
+                applicantId, programId))
+        .flashing("redirected-from-program-slug", programSlug)
```

Comment:
Does this show a toast to applicants? Do we want that?

Reply:
Nope, it's used to show the login modal:

https://github.com/civiform/civiform/blob/c3a918af28eded0b86c794c40ea5b73ae84bd38e/server/app/controllers/applicant/ApplicantProgramReviewController.java#L126-L148

Reply:
Previously we had `.flashing("redirected-from-program-slug", programSlug)` in both cases, so we've just consolidated it in to one method.